src/test/java/net/snowflake/client/jdbc/ResultSetStructuredTypesLatestIT.java

```diff
@@ -204,6 +216,21 @@ public void testReturnAsArrayOfSqlData() throws SQLException {
         });
   }
 
+  @Test
+  public void testReturnAsArrayOfNullableFieldsInSqlData() throws SQLException {
+    SnowflakeObjectTypeFactories.register(SimpleClass.class, FewTypesSqlData::new);
+    withFirstRow(
+        "SELECT OBJECT_CONSTRUCT_KEEP_NULL('string', null, 'nullableIntValue', null, 'nullableLongValue', null, 'longValue', null)::OBJECT(string VARCHAR, nullableIntValue INTEGER, nullableLongValue INTEGER, longValue INTEGER)",
```

Comment:
How `longValue` differs from `nullableLongValue`?

Reply:
I've changed the longValue to primitive type